pkg/metrics/metrics.go

```diff
@@ -93,41 +94,42 @@ func Init(m interface{}, factory Factory, globalTags map[string]string) error {
 					}
 					buckets = append(buckets, b)
 				}
-			} else {
+			default:
 				return fmt.Errorf(
 					"Field [%s]: Buckets should only be defined for Timer and Histogram metric types",
 					field.Name)
 			}
 		}
 		help := field.Tag.Get("help")
 		var obj interface{}
-		if field.Type.AssignableTo(counterPtrType) {
+		switch {
```

Comment:
I tend to prefer the `switch` statement instead of if/else if chains because it's a bit cleaner and easier to read, but others maintainers may disagree.

Reply:
As I started to code with Java I used to be one of those. But different language, different best practices and switch case is one of those differences. Gocritic is widely used which means that the golang community follow it’s rules. If the majority of the developers of this project don’t want to conform to this rule it can be excuded in the golangci-lint config file. 
By the way, I’m not aware of any golang linter forbidding switch cases let me know if you do ;)